lib/test_helpers.coffee

```diff
@@ -20,62 +20,132 @@ class Helpers
+        else
+          !!fs.readFileSync(_path(f), 'utf8').match(regex)
+
+      matches_file: (f, expected, opts = { async: false }) ->
         f = _path(f)
         expected = _path(expected)
-        String(fs.readFileSync(f)) == String(fs.readFileSync(expected))
+        if opts.async
+          W.all([_asyncReadFile(f), _asyncReadFile(expected)])
+            .spread (f, expected) -> f == expected
+        else
+          String(fs.readFileSync(f)) == String(fs.readFileSync(expected))
 
     @directory =
-      is_directory: (dir) ->
-        fs.statSync(_path(dir)).isDirectory()
-      exists: (dir) ->
+      is_directory: (dir, opts = { async: false }) ->
+        if opts.async
+          nodefn.call(fs.stat, _path(dir)).then (s) -> s.isDirectory()
+        else
+          fs.statSync(_path(dir)).isDirectory()
+
+      exists: (dir, opts = { async: false }) ->
         dir = _path(dir)
-        try stat = fs.statSync(dir)
-        catch err then return false
-        stat.isDirectory() and fs.existsSync(dir)
-      doesnt_exist: (dir) ->
-        !fs.existsSync(_path(dir))
-      has_contents: (dir) ->
-        fs.readdirSync(_path(dir)).length > 0
-      is_empty: (dir) ->
-        fs.readdirSync(_path(dir)).length < 1
-      contains_file: (dir, file) ->
-        _.contains(fs.readdirSync(_path(dir)), file)
-      matches_dir: (dir, expected) ->
+        if opts.async
+          nodefn.call(fs.stat, dir)
+            .then (s) -> s.isDirectory()
+            .catch -> false
+        else
+          try stat = fs.statSync(dir)
+          catch then return false
+          stat.isDirectory()
+
+      doesnt_exist: (dir, opts = { async: false }) ->
+        if opts.async
+          _exists(_path(dir), opts).then (exists) -> !exists
+        else
+          !_exists(_path(dir))
+
+      has_contents: (dir, opts = { async: false }) ->
+        if opts.async
+          nodefn.call(fs.readdir, _path(dir)).then (d) -> d.length > 0
+        else
+          fs.readdirSync(_path(dir)).length > 0
+
+      is_empty: (dir, opts = { async: false }) ->
+        if opts.async
+          nodefn.call(fs.readdir, _path(dir)).then (d) -> d.length < 1
+        else
+          fs.readdirSync(_path(dir)).length < 1
+
+      contains_file: (dir, file, opts = { async: false }) ->
+        if opts.async
+          nodefn.call(fs.readdir, _path(dir)).then (d) -> _.contains(d, file)
+        else
+          _.contains(fs.readdirSync(_path(dir)), file)
+
+      matches_dir: (dir, expected, opts = { async: false }) ->
         dir = _path(dir)
         expected = _path(expected)
-        String(fs.readdirSync(dir)) == String(fs.readdirSync(expected))
+        if opts.async
+          W.all([nodefn.call(fs.readdir, dir), nodefn.call(fs.readdir, expected)])
+            .spread (dir, expected) -> String(dir) == String(expected)
+        else
+          String(fs.readdirSync(dir)) == String(fs.readdirSync(expected))
```

Comment:
Considering you're creating type `String`, you should just use the `===` as it won't perform type conversions